lib/tapioca/gem/listeners/mixins.rb

```diff
@@ -71,9 +71,9 @@ def filtered_mixin?(mixin_name)
 
         sig { params(constant: Module).returns(T::Array[Module]) }
         def interesting_ancestors_of(constant)
-          inherited_ancestors_ids = Set.new(
-            inherited_ancestors_of(constant).map { |mod| object_id_of(mod) }
-          )
+          inherited_ancestors = Set.new.compare_by_identity
+          inherited_ancestors.merge(inherited_ancestors_of(constant))
```

Comment:
I think we can do all of this on a single line:

```suggestion
          inherited_ancestors = Set.new.compare_by_identity.merge(inherited_ancestors_of(constant))
```